processor/src/test/resources/fixtures/org/mapstruct/ap/test/nestedbeans/mixed/FishTankMapperImpl.java

```diff
@@ -0,0 +1,358 @@
+
+        fishDto.setKind( fish.getType() );
+
+        return fishDto;
+    }
+
+    protected WaterQualityOrganisationDto waterQualityReportToWaterQualityOrganisationDto(WaterQualityReport waterQualityReport) {
+        if ( waterQualityReport == null ) {
+            return null;
+        }
+
+        WaterQualityOrganisationDto waterQualityOrganisationDto = new WaterQualityOrganisationDto();
+
+        waterQualityOrganisationDto.setName( waterQualityReport.getOrganisationName() );
+
+        return waterQualityOrganisationDto;
+    }
+
+    protected WaterQualityReportDto waterQualityReportToWaterQualityReportDto(WaterQualityReport waterQualityReport) {
+        if ( waterQualityReport == null ) {
+            return null;
+        }
+
+        WaterQualityReportDto waterQualityReportDto = new WaterQualityReportDto();
+
+        waterQualityReportDto.setOrganisation( waterQualityReportToWaterQualityOrganisationDto( waterQualityReport ) );
+        waterQualityReportDto.setVerdict( waterQualityReport.getVerdict() );
+
+        return waterQualityReportDto;
+    }
+
+    protected WaterQualityDto waterQualityToWaterQualityDto(WaterQuality waterQuality) {
+        if ( waterQuality == null ) {
+            return null;
+        }
+
+        WaterQualityDto waterQualityDto = new WaterQualityDto();
+
+        waterQualityDto.setReport( waterQualityReportToWaterQualityReportDto( waterQuality.getReport() ) );
+
+        return waterQualityDto;
+    }
+
+    private Ornament sourceInteriorOrnament(FishTank fishTank) {
+
+        if ( fishTank == null ) {
+            return null;
+        }
+        Interior interior = fishTank.getInterior();
+        if ( interior == null ) {
+            return null;
+        }
+        Ornament ornament = interior.getOrnament();
+        if ( ornament == null ) {
+            return null;
+        }
+        return ornament;
+    }
+
+    protected OrnamentDto ornamentToOrnamentDto(Ornament ornament) {
+        if ( ornament == null ) {
+            return null;
+        }
+
+        OrnamentDto ornamentDto = new OrnamentDto();
+
+        ornamentDto.setType( ornament.getType() );
+
+        return ornamentDto;
+    }
+
+    protected WaterPlantDto waterPlantToWaterPlantDto(WaterPlant waterPlant) {
+        if ( waterPlant == null ) {
+            return null;
+        }
+
+        WaterPlantDto waterPlantDto = new WaterPlantDto();
+
+        waterPlantDto.setKind( waterPlant.getKind() );
+
+        return waterPlantDto;
+    }
+
+    private Ornament sourceInteriorOrnament1(FishTank fishTank) {
+
+        if ( fishTank == null ) {
+            return null;
+        }
+        Interior interior = fishTank.getInterior();
+        if ( interior == null ) {
+            return null;
+        }
+        Ornament ornament = interior.getOrnament();
+        if ( ornament == null ) {
+            return null;
+        }
+        return ornament;
+    }
+
+    protected Fish fishDtoToFish(FishDto fishDto) {
```

Comment:
The `private` methods are the methods generated by the `NestedPropertyMappingMethod`, in [`NestedPropertyMappingMethod.ftl#L21`](https://github.com/mapstruct/mapstruct/blob/master/processor/src/main/resources/org/mapstruct/ap/internal/model/NestedPropertyMappingMethod.ftl#L21) we explicitly use `private`. The other methods are actually forged `BeanMappingMethod`(s) which use the accessibility from a `ForgedMethod` which actually is `protected`. If you want we can put this in #1082 as well.